components/forms/input-fields/input-field.scss

```diff
@@ -2,167 +2,152 @@
     }
-
-    &--small {
-      padding-left: 12px;
-      height: 36px;
-      font-size: 12px;
-      line-height: 18px;
-      padding-top: 12px;
-
-      &:focus ~ label {
-        left: 13px;
-        top: 5px;
+    &__text-input {
+      font-size: 16px;
+      color: $gray-10;
+      height: 56px;
+      width: 100%;
+      padding-left: 20px;
+      border: 1px solid $gray-5;
+      border-radius: 2px;
+      padding-top: 13px;
+      &::placeholder {
+        color: transparent;
+      }
+      &:read-only {
+        pointer-events: none;
+        background: $gray-3;
+        color: $black;
       }
-
-      & ~ label {
-        left: 13px;
-        font-size: 12px;
-        top: calc(50% - 20px);
+      &:read-only + label {
+        pointer-events: none;
+        color: $gray-7;
       }
-
-      &:not(:placeholder-shown) ~ label {
-        top: 5px;
+      &:disabled {
+        background: $gray-3;
+        & ~ label {
+          color: $gray-7;
+        }
       }
-    }
-
-    &--disabled {
-      &:hover {
-        background-color: $gray-3;
+      &:focus {
+        outline: 4px solid rgba(0, 153, 218, 0.5);
+        border: 1px solid $clear-blue;
       }
-
-      & ~ label {
-        color: $gray-7;
+      &:focus + label {
+        top: 10px;
+        font-size: 10px;
+      }
+      &:not(:placeholder-shown) + label {
+        top: 10px;
+        font-size: 10px;
       }
-      background: $gray-3;
     }
-
-    &--readonly {
-      color: $black;
+    &__label {
+      position: absolute;
+      top: 18px;
+      left: 20px;
+      transition: 0.5s;
       pointer-events: none;
-      & ~ label {
-        color: $black;
+      color: $gray-8;
+      font-size: 16px;
+    }
+    &__info-message {
+      font-size: 12px;
+      line-height: 18px;
+      color: $gray-9;
+      margin-left: 3px;
+      margin-top: 6px;
+    }
+    &--small {
+      .msds-input__text-input {
+        padding-left: 12px;
+        height: 36px;
+        font-size: 12px;
+        line-height: 18px;
+        padding-top: 12px;
+        top: 5px;
+        & + label {
+          left: 13px;
+          font-size: 12px;
+          top: 10px;
+        }
+        &:focus + label {
+          left: 13px;
+          top: 1px;
+          font-size: 11px;
+        }
+        &:not(:placeholder-shown) + label {
+          font-size: 11px;
+          top: 1px;
+        }
       }
     }
-
     &--success {
-      border: 1px solid $confirm-green;
-      & ~ p {
-        color: $confirm-green;
-      }
-
-      & ~ p:before,
-      & ~ p:after {
+      .msds-input__icon-wrapper:before,
+      .msds-input__icon-wrapper:after {
         content: '';
         position: absolute;
         right: 15px;
+        top: calc(50% - 6px);
+        height: 13px;
         width: 2px;
         background-color: $confirm-green;
       }
-
-      & ~ p:before {
-        top: calc(50% - 16px);
+      .msds-input__icon-wrapper:before {
+        top: calc(50% - 7px);
```

Comment:
Wasn't it this calc position that was causing the bug with the text and icon?

Reply:
I can't hardcode it unfortunately as it is tied to the hight of the input field which can vary as we have 2 different sizes for it. else I have to create a separate markup and CSS for the small input class